app/users/index/controller.js

```diff
@@ -1,106 +1,101 @@
 import Controller from "@ember/controller";
-import { computed, action } from "@ember/object";
-import { reads } from "@ember/object/computed";
+import { action, set } from "@ember/object";
 import { inject as service } from "@ember/service";
-import { task, timeout, hash } from "ember-concurrency";
-import QueryParams from "ember-parachute";
+import { tracked } from "@glimmer/tracking";
+import { restartableTask, timeout, hash } from "ember-concurrency";
+import { task as trackedTask } from "ember-resources/util/ember-concurrency";
 import moment from "moment";
 
-const UsersQueryParams = new QueryParams({
-  search: {
-    defaultValue: "",
-    replace: true,
-    refresh: true,
-  },
-  supervisor: {
-    defaultValue: null,
-    replace: true,
-    refresh: true,
-  },
-  active: {
-    defaultValue: "1",
-    replace: true,
-    refresh: true,
-  },
-  ordering: {
-    defaultValue: "username",
-    replace: true,
-    refresh: true,
-  },
-});
-
-const UsersIndexController = Controller.extend(UsersQueryParams.Mixin, {
-  session: service("session"),
-  router: service("router"),
-
-  currentUser: reads("session.data.user"),
-
-  viewUserProfile: action(function (userId) {
-    return this.router.transitionTo("users.edit", userId);
-  }),
-
-  selectedSupervisor: computed(
-    "prefetchData.lastSuccessful.value.supervisor",
-    "store",
-    "supervisor",
-    function () {
-      return this.supervisor && this.store.peekRecord("user", this.supervisor);
-    }
-  ),
-
-  setup() {
+export default class UsersIndexController extends Controller {
+  queryParams = ["search", "supervisor", "active", "ordering"];
+  @service session;
+  @service router;
+  @service store;
+  @tracked currentUser = this.session.data.user;
```

Comment:
This is not the same as `currentUser: reads("session.data.user")`

```suggestion
  get currentUser  () {
    return this.session.data.user
  }
```